subsys/bootloader/CMakeLists.txt

```diff
@@ -177,13 +169,32 @@ add_custom_target(
   offsets_h
   )
 
+if(CONFIG_SOC_NRF52840)
+	set(ADDR_OVERRIDE
+		--s0-addr 16384
+		--s1-addr 65536
+		--provision-addr 1044480)
+elseif(CONFIG_SOC_NRF52832)
+	set(ADDR_OVERRIDE
+		--s0-addr 16384
+		--s1-addr 65536
+		--provision-addr 520192)
+elseif(CONFIG_SOC_NRF9160)
+	set(ADDR_OVERRIDE
+		--s0-addr 32768
+		--s1-addr 786432
+		--provision-addr 1044480)
+endif()
+
+
```

Comment:
DT_FLASH_AREA_S0_OFFSET is now available in the DT namespace (you can do ${DT_FLASH_AREA_S0_OFFSET}).

So I think you want to remove the generated_dts_board.conf parser function, and instead determine the possibly overridden value in CMake. instead of in python. Ask me for details. I could be wrong.

Reply:
It's a great idea unfortuntaly these symbols does not yet exsists in the DT https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/1d6219ffec845a43f66469354d997fbea2f283a4/boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts#L165

I'll investigate more into your 2nd comment I'll ask you tomorrow :)

Reply:
I can't read DT source code and predict what symbols will be available ;)

The reason I believed they were available is that when I build, I get symbols like these in CMake:

```
DT_FLASH_AREA_MCUBOOT_OFFSET=0
DT_FLASH_AREA_STORAGE_OFFSET=122880

```

But I haven't tried in a build that has the S0 area defined.

Reply:
Trying to predict naming and which symbols are available from DT is quite hard and it was a nice guess the only thing missing to get the symbols you are asking for is 

```
storage_partition: partition@(offset) { //@has no effect and is just inferred as a comment)
			label = "s0";
			reg = <(offset) (size)>;
};
```
in the partition section but this will have to be added for every board instead of every SoC I think? Honestly I'm not sure what would be best, this would at least give the smallest diff towards zephyr if we don't want to try to upstream partitioning changes upstream.

A problem with trying to upstream it is that MCUBoot is relaying on the mcuboot symbols as they can't overlay their own DT overlay file outside the zephyr project folder as overlays have to be board specific to be overlayed

Reply:
Moved the code out of python and into cmake now so that instead of an override we do an fallback if the values does not exsist in DT

Reply:
Great!